lodash.js

```diff
@@ -9903,6 +9931,40 @@
       return (result == '0' && (1 / value) == -INFINITY) ? '-0' : result;
     }
 
+    /**
+     * Converts `value` to a number if it's not one.
+     * **Note:** Loosely based on [`ToNumber`](http://www.ecma-international.org/ecma-262/6.0/#sec-tonumber)
+     *
+     * @static
+     * @memberOf _
+     * @category Lang
+     * @param {*} value The value to process.
+     * @returns {number} Returns the number.
+     */
+    function toNumber(value) {
+      if (!isNumber(value)) {
+        if (isObject(value) && isFunction(value.valueOf)) {
+          value = value.valueOf.call(value);
+        }
+        if (isObject(value)) {
+          value = toString(value);
+        }
+        if (isString(value)) {
+          value = trim(value);
+          if (reIsBinary.test(value)) {
+            return nativeParseInt(value.slice(2), 2);
+          }
+          if (reIsOctal.test(value)) {
+            return nativeParseInt(value.slice(2), 8);
+          }
+          if (reIsBadHex.test(value)) {
+            return NaN;
+          }
+        }
+      }
+      return +value;
+    }
+
```

Comment:
That is going to fail on at least node v0.8 I believe. But that should just be a case of adding the bad hex test back in. It's nice to see where your experience pays off wrt simplification. Let me change it and we can see.